models/applicantModel.js

```diff
@@ -1,28 +1,461 @@
+        {
+          router_index,
+          origin_country_name,
+          origin_city_name,
+          destination_country_name,
+          destination_city_name,
+          beginning_date,
+          beginning_time,
+          ending_date,
+          ending_time,
+          plane_needed,
+          hotel_needed,
+        },
+        additionalRoutes,
+      );
+
+      // =======================================
+      // Step 1: Insert into Request table
+      // =======================================
+      const request_days = getRequestDays(allRoutes);
+
+      // Get Status from role
+      const role = await conn.query(
+        `SELECT role_id FROM user WHERE user_id = ?`,
+        [user_id],
+      );
+
+      console.log("Role ID:", role[0].role_id);
+      let request_status;
+      if (role[0].role_id == 1) {
+        console.log("Role ID:", role[0].role_id);
+        request_status = 1; // 1 = Open
+      }
+      else if (role[0].role_id == 4) {
+        console.log("Role ID:", role[0].role_id);
+        request_status = 2; // 2 = First Review
+      }
+      else if (role[0].role_id == 5) {
+        console.log("Role ID:", role[0].role_id);
+        request_status = 3; // 3 = Second Review
+      }
+      else {
+        throw new Error("User role in not allowed to create a travel request");
+      }
+
+      const insertIntoRequestTable = `
+        INSERT INTO Request (
+          user_id, request_status_id, notes, requested_fee, imposed_fee, request_days
+        ) VALUES (?, ?, ?, ?, ?, ?)
+      `;
+
+      const requestTableResult = await conn.execute(insertIntoRequestTable, [
+        user_id,
+        request_status,
+        notes,
+        requested_fee,
+        imposed_fee,
+        request_days,
+      ]);
+
+      const requestId = requestTableResult.insertId;
+
+      // =======================================
+      // Step 2: Insert into Country & City table
+      // =======================================
+
+      for (const route of allRoutes) {
+        try {
+          console.log("Processing route:", route);
+
+          let
+            id_origin_country,
+            id_destination_country,
+            id_origin_city,
+            id_destination_city;
+
+          // Search if the country exists in the database
+          id_origin_country = await getCountryId(conn, route.origin_country_name);
+          id_destination_country = await getCountryId(conn, route.destination_country_name);
+
+          console.log("Country IDs:", id_origin_country, id_destination_country);
+
+          // Search if the city exists in the database
+          id_origin_city = await getCityId(conn, route.origin_city_name);
+          id_destination_city = await getCityId(conn, route.destination_city_name);
+
+          console.log("City IDs:", id_origin_city, id_destination_city);
+          // Insert into Route table
+
+          const insertRouteTable = `
+          INSERT INTO route (
```

Comment:
```suggestion
          INSERT INTO Route (
```
Name of the table is incorrect.